jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -261,6 +284,27 @@ def _jwst2data(file_obj, ext, data_label):
 
     return data, new_data_label
 
+# ---- Functions that handle input from Roman ASDF files -----
+
+
+def _roman_2d_asdf_to_glue_data(file_obj, data_label):
```

Comment:
This auto loading of all the extensions is problematic.

1. Performance: If you do not really need the data, do not load it. Otherwise, Glue will try to link them all and the way linking works, you will see lag very quickly if for each file, you link 6 things.
2. This function should respect the `ext` that user passes it from high-level API and act similarly to parsers of other formats. That is, if `ext=None`, you load only DATA (90% of people only cares about that one). If `ext` is one of the other ones, you load only that one. If `ext='*'`, then you load all the 2D extensions.